### ISSUE TITLE
Log Group retention policy update not working

### ISSUE THREAD

**Issue by transadm312** [aws/amazon-cloudwatch-logs-for-fluent-bit#254]
when creating Log Group for first time retention updated correctly
after somedays I updated log retention policy..but retention didn't update in AWS CloudWatch ..

I suspect it didn't reach below line https://github.com/aws/amazon-cloudwatch-logs-for-fluent-bit/blob/mainline/cloudwatch/cloudwatch.go#L593 and return from https://github.com/aws/amazon-cloudwatch-logs-for-fluent-bit/blob/mainline/cloudwatch/cloudwatch.go#L589

because "Log group already exists" is not printing in log

**Comment by zhonghui12** [comment]
Thanks for your reporting. I think it should still work if you enable the `auto_create_group`, have you tried it?

**Comment by amosaini** [comment]
yes It is enabled for us. We are also facing the same. The put retention policy is not working when the log group already exist. 
<img width="198" alt="image" src="https://user-images.githubusercontent.com/95207359/178438329-f476e2c8-00ae-4b49-a8a5-dba02c3bb1ab.png">

**Comment by abhijit-gaonkar** [comment]
@zhonghui12 was this issue fixed?
